src/itensors.jl

```diff
@@ -41,3 +41,20 @@ trivial_space(x::Index) = _trivial_space(x)
 trivial_space(x::Vector{<:Index}) = _trivial_space(x)
 trivial_space(x::ITensor) = trivial_space(inds(x))
 trivial_space(x::Tuple{Vararg{Index}}) = trivial_space(first(x))
+
+"""
+Given an input tensor and a Dict (ind_to_newind), replace inds of tensor that are also
+keys of ind_to_newind to the value of ind_to_newind.
+Note that it is the same as
+ITensors.replaceinds(tensor, collect(keys(ind_to_newind)) => collect(values(ind_to_newind))).
+Based on benchmark, this implementation is more efficient when the size of ind_to_newind is large.
+TODO: we can remove this function once the original replaceinds performance is improved.
+"""
+function ITensors.replaceinds(tensor::ITensor, ind_to_newind::Dict{<:Index,<:Index})
```

Comment:
Could you raise an issue in `ITensors.jl` reminding us to fix this over there?

Reply:
SG, will do

Reply:
ITensors.jl issue: https://github.com/ITensor/ITensors.jl/issues/1078